### ISSUE TITLE
bug: passing predictions instead of logits to tensorflow functions

### ISSUE THREAD

**Issue by zsal** [blei-lab/edward#795]
I believe there is a bug in evaluate.py, where we pass probability predictions (values between 0 and 1) to tensorflow functions that expect logits.

At a minimum I would suggest at least doing something like following (not 100% sure on the tensorflow syntax...). More preferably, we can pass logits directly to this function rather then doing a round trip through a sigmoid function and its inverse.

```
def binary_crossentropy(y_true, y_pred):
  """Binary cross-entropy.

  Parameters
  ----------
  y_true : tf.Tensor
    Tensor of 0s and 1s.
  y_pred : tf.Tensor
    Tensor of real values (logit probabilities), with same shape as
    ``y_true``.
  """
  y_true = tf.cast(y_true, tf.float32)
  y_pred = tf.cast(y_pred, tf.float32)
  y_logit = tf.log(y_pred) - tf.log(tf.subtract(1.0, y_pred))
  return tf.reduce_mean(
      tf.nn.sigmoid_cross_entropy_with_logits(logits=y_logit, labels=y_true))
```

Note: the issue seems to be ubiquitous for any "with_logits" tf function in evaluate.py.

**Comment by dustinvtran** [comment]
Thanks for catching this. For now, I think the easiest hack is as you suggest, which is to sigmoid-squash the probabilities. Right now in `ed.evaluate`'s inputs/outputs API, there's no way to pass in `logits`. Thoughts are welcome.
